src/core/portal/dom-portal-host.ts

```diff
@@ -0,0 +1,56 @@
+import {DynamicComponentLoader, AppViewManager, ComponentRef} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {BasePortalHost, ComponentPortal, TemplatePortal} from './portal';
+
+
+/**
+ * A PortalHost for attaching portals to an arbitrary DOM element outside of the Angular
+ * application context.
+ *
+ * This is the only part of the portal core that directly touches the DOM.
+ */
+export class DomPortalHost extends BasePortalHost {
+  constructor(
+      private _hostDomElement: Element,
+      private _componentLoader: DynamicComponentLoader,
+      private _viewManager: AppViewManager) {
+    super();
+  }
+
+  /** Attach the given ComponentPortal to DOM element using the DynamicComponentLoader. */
+  attachComponentPortal(portal: ComponentPortal): Promise<ComponentRef> {
+    if (portal.origin == null) {
+      throw new BaseException(
```

Comment:
We can chat about it on Monday. I don't yet see the end-of-the-day value in having different error types, but I'm open to convincing.